src/intro_movie_scenes_lib.c

```diff
@@ -0,0 +1,528 @@
+#include "global.h"
+#include "intro_movie_internal.h"
+#include "math_util.h"
+#include "system.h"
+#include "sys_task_api.h"
+#include "unk_0200FA24.h"
+
+void Task_IntroMovie_BlendFadeEffect(SysTask *task, void *pVoid);
+int IntroMovie_BgLayerToScrollEffectSlot(enum GFBgLayer bgId);
+void Task_IntroMovie_BgScroll_VBlank(SysTask *task, void *pVoid);
+void Task_IntroMovie_BgScroll_NotVBlank(SysTask *task, void *pVoid);
+void Task_IntroMovie_WindowPanEffect(SysTask *task, void *pVoid);
+void Task_IntroMovie_CircleWipeEffect(SysTask *task, void *pVoid);
+void IntroMovie_CircleWipeEffect_HBlankCB(void *pVoid);
+
+void IntroMovie_CreateSpriteResourceManagers(IntroMovieOvyData *data, const u8 *counts) {
+    for (u8 i = 0; i < 4; ++i) {
+        data->spriteResManagers[i] = Create2DGfxResObjMan(counts[i], (GfGfxResType)i, HEAP_ID_INTRO_MOVIE);
+    }
+}
+
+void IntroMovie_DestroySpriteResourceManagers(IntroMovieOvyData *data) {
+    for (u8 i = 0; i < 4; ++i) {
+        Destroy2DGfxResObjMan(data->spriteResManagers[i]);
+    }
+}
+
+GF_2DGfxResMan **IntroMovie_GetSpriteResourceManagersArray(IntroMovieOvyData *data) {
+    return data->spriteResManagers;
+}
+
+void IntroMovie_StartSpriteAnimAndMakeVisible(Sprite *sprite, BOOL active) {
+    Set2dSpriteAnimActiveFlag(sprite, active);
+    Set2dSpriteVisibleFlag(sprite, active);
+}
+
+void IntroMovie_BuildSpriteResourcesHeaderAndTemplate(int resId, IntroMovieOvyData *data, int priority, NNS_G2D_VRAM_TYPE whichScreen, SpriteTemplate *template, SpriteResourcesHeader *header) {
+    CreateSpriteResourcesHeader(
+        header,
+        resId,
+        resId,
+        resId,
+        resId,
+        -1,
+        -1,
+        0,
+        priority,
+        data->spriteResManagers[GF_GFX_RES_TYPE_CHAR],
+        data->spriteResManagers[GF_GFX_RES_TYPE_PLTT],
+        data->spriteResManagers[GF_GFX_RES_TYPE_CELL],
+        data->spriteResManagers[GF_GFX_RES_TYPE_ANIM],
+        NULL,
+        NULL
+    );
+    template->spriteList = data->spriteList;
+    template->header = header;
+    SetVec(template->position, 0, 0, 0);
+    SetVec(template->scale, FX32_ONE, FX32_ONE, FX32_ONE);
+    template->rotation = 0;
+    template->priority = 0;
+    template->whichScreen = whichScreen;
```

Comment:
I'm still not sure if this should be whichScreen

Reply:
we can change it later when we gain new confidence on what it should be